Error in conditional probabilities with 0 entries

See #1512 for more context

```python
        demography = msprime.Demography()
        demography.add_population(name="A", initial_size=1)
        demography.add_population(name="B", initial_size=1)
        demography.add_population(name="C", initial_size=1)
        demography.add_admixture(
                time=1, ancestral=["B", "C"], proportions=[1.0, 0], derived="A")
        debug = demography.debug()
        lineages = debug.possible_lineage_locations(["A", "B", "C"])
```
gives
```
tests/test_demography.py:3809: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
msprime/demography.py:2962: in possible_lineage_locations
    P_out = self.lineage_probabilities(all_steps, sample_time=sample_time)
msprime/demography.py:2872: in lineage_probabilities
    for lm in demo._as_lineage_movements():
msprime/demography.py:2373: in _as_lineage_movements
    S = _proportions_to_sequential(self.proportions)
msprime/demography.py:2497: in _proportions_to_sequential
    C = [P[j] / (1 - sum(P[:j])) for j in range(len(P))]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <range_iterator object at 0x7f95054f4810>

>   C = [P[j] / (1 - sum(P[:j])) for j in range(len(P))]
E   ZeroDivisionError: float division by zero

msprime/demography.py:2497: ZeroDivisionError
```

~@apragsdale, can I leave this one with you? Should be an easy fix. I've skipped the ``test_admixture_zero_proportion`` test, which you can unskip.~ Actually it's an easy fix, I'll sort it out.

[Edit] Oops, turns out this is nothing to do with the lineage location code.

--- a/msprime/demography.py
+++ b/msprime/demography.py
@@ -2490,7 +2490,11 @@ def _proportions_to_sequential(P):
     # Conditioned on having already distributed a fraction q of the
     # lineages, the we need a fraction p / (1 - q) of the remaining
     # lineages to get an overall proportion of p
-    C = [P[j] / (1 - sum(P[:j])) for j in range(len(P))]
+    C = [0 for _ in P]
+    for j in range(len(P)):
+        s = sum(P[:j])
+        if s < 1:
+            C[j] = P[j] / (1 - s)
     return C
 
 

--- a/tests/test_demography.py
+++ b/tests/test_demography.py
@@ -5253,6 +5253,31 @@ class TestNormaliseLineageMovements:
         }
         assert equiv == demography._normalise_lineage_movements(demography.events)
 
+    def test_admixture_zero_proportion(self):
+        demography = msprime.Demography.isolated_model([10] * 10)
+        demography.add_admixture(
+            time=0.1, derived=0, ancestral=[1, 2], proportions=[0, 1]
+        )
+        equiv = {
+            0: [
+                msprime.LineageMovement(0, 1, 0),
+                msprime.LineageMovement(0, 2, 1),
+            ]
+        }
+        assert equiv == demography._normalise_lineage_movements(demography.events)
+
+        demography = msprime.Demography.isolated_model([10] * 10)
+        demography.add_admixture(
+            time=0.1, derived=0, ancestral=[1, 2], proportions=[1, 0]
+        )
+        equiv = {
+            0: [
+                msprime.LineageMovement(0, 1, 1),
+                msprime.LineageMovement(0, 2, 0),
+            ]
+        }
+        assert equiv == demography._normalise_lineage_movements(demography.events)
+
 
 class TestProportionConversion:
     """
@@ -5284,7 +5309,12 @@ class TestProportionConversion:
         "P",
         [
             [1],
+            [0, 1],
+            [1, 0],
             [1 / 2, 1 / 2],
+            [1 / 2, 1 / 2, 0],
+            [1 / 2, 0, 1 / 2],
+            [0, 1 / 2, 1 / 2],
             [1 / 3] * 3,
             [1 / 6] * 6,
             [2 / 6, 1 / 6, 2 / 6, 1 / 6],

